### ISSUE TITLE
Scope of  member variable  (private final URL res) in XtextTemplateStore needs to be changed to protected

### ISSUE THREAD

**Issue by srinivasiyermv** [eclipse/xtext-eclipse#1561]
**Problem** :  Extending XtextTemplateStore  for modifications is restricted due to  scope of member variable **res** is tightly coupled  with loadContributedTemplates().

```
private final URL res;

@Override
	protected void loadContributedTemplates() throws IOException {
  ...
         openStream = res.openStream();
  ..
  ..
 }
```

**Comment by kthoms** [comment]
The value of `res` is assigned from `getTemplateFileURL()`. Would overriding this method serve your requirement?

**Comment by cdietrich** [comment]
res beeing protected or having a getter would avoid large scale c&p

**Comment by kthoms** [comment]
I'd prefer a protected getter then. I do not see that the field should be modifiable after initialization.

@srinivasiyermv Would you be so nice to provide a patch? See our [Contribution guide](https://github.com/eclipse/xtext/blob/master/CONTRIBUTING.md) and feel free to ask.

**Comment by srinivasiyermv** [comment]
Sure will create a patch for the same.. Thanks.

**Comment by kthoms** [comment]
Please add `@since 2.24` to the getter then

**Comment by cdietrich** [comment]
Fixed in 2.24
